<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 9 Testing | A Guide to The New Statistics with R</title>
<meta name="author" content="Pete Apicella">
<meta name="generator" content="bookdown 0.27 with bs4_book()">
<meta property="og:title" content="Chapter 9 Testing | A Guide to The New Statistics with R">
<meta property="og:type" content="book">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 9 Testing | A Guide to The New Statistics with R">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
<meta name="description" content="9.1 Significance testing: Time for t Statisticians think that the overfixation on p-values has contributed to the reproducibility crisis in science That’s why the author’s focus of this text is on...">
<meta property="og:description" content="9.1 Significance testing: Time for t Statisticians think that the overfixation on p-values has contributed to the reproducibility crisis in science That’s why the author’s focus of this text is on...">
<meta name="twitter:description" content="9.1 Significance testing: Time for t Statisticians think that the overfixation on p-values has contributed to the reproducibility crisis in science That’s why the author’s focus of this text is on...">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">A Guide to The New Statistics with R</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">About</a></li>
<li><a class="" href="introduction.html"><span class="header-section-number">1</span> Introduction</a></li>
<li><a class="" href="motivation.html"><span class="header-section-number">2</span> Motivation</a></li>
<li><a class="" href="description.html"><span class="header-section-number">3</span> Description</a></li>
<li><a class="" href="reproducible-research.html"><span class="header-section-number">4</span> Reproducible Research</a></li>
<li><a class="" href="estimation.html"><span class="header-section-number">5</span> Estimation</a></li>
<li><a class="" href="linear-models.html"><span class="header-section-number">6</span> Linear models</a></li>
<li><a class="" href="regression.html"><span class="header-section-number">7</span> Regression</a></li>
<li><a class="" href="prediction.html"><span class="header-section-number">8</span> Prediction</a></li>
<li><a class="active" href="testing.html"><span class="header-section-number">9</span> Testing</a></li>
<li><a class="" href="intervals.html"><span class="header-section-number">10</span> Intervals</a></li>
<li><a class="" href="analysis-of-variance.html"><span class="header-section-number">11</span> Analysis of Variance</a></li>
<li><a class="" href="factorial-designs.html"><span class="header-section-number">12</span> Factorial designs</a></li>
<li><a class="" href="analysis-of-covariance.html"><span class="header-section-number">13</span> Analysis of Covariance</a></li>
<li><a class="" href="linear-model-complexities.html"><span class="header-section-number">14</span> Linear Model Complexities</a></li>
<li><a class="" href="generalized-linear-models-glms.html"><span class="header-section-number">15</span> Generalized Linear Models (GLMs)</a></li>
<li><a class="" href="glms-for-count-data.html"><span class="header-section-number">16</span> GLMs for Count Data</a></li>
<li><a class="" href="binomial-glms.html"><span class="header-section-number">17</span> Binomial GLMs</a></li>
<li><a class="" href="glms-for-binary-data.html"><span class="header-section-number">18</span> GLMs for binary data</a></li>
<li><a class="" href="conclusions.html"><span class="header-section-number">19</span> Conclusions</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/apicellap/NSwR.git">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="testing" class="section level1" number="9">
<h1>
<span class="header-section-number">9</span> Testing<a class="anchor" aria-label="anchor" href="#testing"><i class="fas fa-link"></i></a>
</h1>
<div id="significance-testing-time-for-t" class="section level2" number="9.1">
<h2>
<span class="header-section-number">9.1</span> Significance testing: Time for t<a class="anchor" aria-label="anchor" href="#significance-testing-time-for-t"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Statisticians think that the overfixation on p-values has contributed to the reproducibility crisis in science</li>
<li>That’s why the author’s focus of this text is on estimation, which relies on estimates and confidence intervals</li>
<li>This chapter introduces the Student’s t-test which will generate a p-value</li>
</ul>
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"arm"</span>,  repos <span class="op">=</span> <span class="st">"https://cran.us.r-project.org"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"ggplot2"</span>,  repos <span class="op">=</span> <span class="st">"https://cran.us.r-project.org"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"Sleuth2"</span>,  repos <span class="op">=</span> <span class="st">"https://cran.us.r-project.org"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"SMPracticals"</span>,  repos <span class="op">=</span> <span class="st">"https://cran.us.r-project.org"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb106"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://CRAN.R-project.org/package=arm">arm</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://r-forge.r-project.org/projects/sleuth2/">Sleuth2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://statwww.epfl.ch/davison/SM/">SMPracticals</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="students-t-test-darwins-maize" class="section level2" number="9.2">
<h2>
<span class="header-section-number">9.2</span> Student’s t-test: Darwin’s maize<a class="anchor" aria-label="anchor" href="#students-t-test-darwins-maize"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>The Student’s t-test
<ul>
<li>Uses the t-distribution
<ul>
<li>This is like a small sample size version of the normal distribution</li>
</ul>
</li>
<li>There are two basic forms a t-test can take:
<ul>
<li>The one sample t-test takes the mean of a sample and compares it with a null hypothesis of zero</li>
<li>The two sample t-test compares the difference between the means of two samples against a null a hypothesis of no difference (zero)
<ul>
<li>The paired two sample t-test is a subtype that applies when the values of two samples come in pairs (like in the darwin maize data)</li>
</ul>
</li>
</ul>
</li>
<li>R has a function for the t-test but the author tends to avoid it. Criticism from the author about the default R t-test:
<ul>
<li>the parts of the t-test include:
<ul>
<li>a difference</li>
<li>its standard error (and the CI used to calculate it)</li>
<li>the observed value of t</li>
<li>the critical value of t</li>
<li>the degrees of freedom</li>
<li>the P-value</li>
</ul>
</li>
<li>the default R t-test output is poorly ordered and missing some of this information</li>
<li>the default is not the classic t-test but another variant called Welch’s
<ul>
<li>Welch’s might be good for research since it doesn’t assume equal variance though</li>
</ul>
</li>
<li>but adds another twist so it’s not appropriate for teaching beginners</li>
</ul>
</li>
</ul>
</li>
<li>the general form of a t-test is:</li>
</ul>
<center>
<ul>
<li><span class="math inline">\(observed \;t = \frac{difference}{SE}\)</span></li>
</ul>
</center>
<ul>
<li>this is a more formal version of what was done in earlier chapters in which the output of the display() function - the estimates and the standard errors were compared</li>
<li>as a rule of thumb/eyeball test, the estimate should be twice as large as its standard error to reject the null hypothesis at the lowest level of confidence (95%) and three times as large for the next level (99% CI) and so on</li>
<li>when sample sizes are larger, t converges to the normal distribution</li>
<li>when sample sizes are smaller, the t distribution is shorter and wider than normal</li>
</ul>
<div class="inline-figure"><img src="fig9-1.png" width="65%"></div>
<ul>
<li>two t values of the t-test:
<ul>
<li>critical t value - sets the bar for comparison - this is the minimum t value required to achieve a given level of significance (P = 0.05, 0.01, etc.)</li>
<li>observed t value - calculated by dividing the estimate by its standard error
<ul>
<li>if the observed t value is larger than the critical t value, then the result is declared significant at that level</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="inline-figure"><img src="fig9-2.png" width="65%"></div>
<ul>
<li>For performing a paired t-test with darwin’s maize data, the observed t value must be compared to the critical value with the corresponding critical t value for the sample size that Darwin had (<span class="math inline">\(n = 15 \;pairs\)</span>)</li>
</ul>
<p>Calculate the critical t value when P = 0.05 and n = 15 pairs:</p>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/TDist.html">qt</a></span><span class="op">(</span><span class="fl">0.975</span>, df <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>  <span class="co">#mean + 2SEs; 0.975 is the upper CI limit for a 95% CI </span>
<span class="co">#&gt; [1] 2.144787</span></code></pre></div>
<p>Summarize the model for a version of the maize data that omits the pairing aspect (for standard two sample t-test):</p>
<div class="sourceCode" id="cb108"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">height</span> <span class="op">~</span> <span class="va">type</span>, data <span class="op">=</span> <span class="va">darwin</span><span class="op">)</span><span class="op">)</span> <span class="co">#this linear model omits the pairing aspect </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:</span>
<span class="co">#&gt; lm(formula = height ~ type, data = darwin)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Residuals:</span>
<span class="co">#&gt;     Min      1Q  Median      3Q     Max </span>
<span class="co">#&gt; -8.1917 -1.0729  0.8042  1.9021  3.3083 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Coefficients:</span>
<span class="co">#&gt;             Estimate Std. Error t value Pr(&gt;|t|)    </span>
<span class="co">#&gt; (Intercept)  20.1917     0.7592  26.596   &lt;2e-16 ***</span>
<span class="co">#&gt; typeSelf     -2.6167     1.0737  -2.437   0.0214 *  </span>
<span class="co">#&gt; ---</span>
<span class="co">#&gt; Signif. codes:  </span>
<span class="co">#&gt; 0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Residual standard error: 2.94 on 28 degrees of freedom</span>
<span class="co">#&gt; Multiple R-squared:  0.175,  Adjusted R-squared:  0.1455 </span>
<span class="co">#&gt; F-statistic:  5.94 on 1 and 28 DF,  p-value: 0.02141</span></code></pre></div>
<ul>
<li>
<code><a href="https://rdrr.io/r/base/summary.html">summary()</a></code> takeaways:
<ul>
<li>The second row of the coefficients table tests the null hypothesis by comparing the observed difference in height between progeny of selfed and cross pollinated plants (-2.6167)</li>
<li>The first row tests the mean height of the cross pollinated plants versus a null hypothesis of an average height of zero
<ul>
<li>This was not a comparison we intended to make in advance or really at all</li>
</ul>
</li>
</ul>
</li>
<li>It can be beneficial to specify only the tests we want to have run even if it means more typing</li>
<li>The average difference in height is -2.6167 and its standard error of the difference is 1.0737</li>
</ul>
<p>This produces an observed t value of:</p>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">-</span><span class="fl">2.6167</span><span class="op">/</span><span class="fl">1.0737</span> <span class="co">#difference/standard error of the difference </span>
<span class="co">#&gt; [1] -2.437087</span></code></pre></div>
<ul>
<li>critical t value for this data is 2.144787</li>
<li>observed t value for the data is -2.437087</li>
</ul>
<p>It is better/more informative to work with CIs:</p>
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/confint.html">confint</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">height</span> <span class="op">~</span> <span class="va">type</span>, data <span class="op">=</span> <span class="va">darwin</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;                2.5 %     97.5 %</span>
<span class="co">#&gt; (Intercept) 18.63651 21.7468231</span>
<span class="co">#&gt; typeSelf    -4.81599 -0.4173433</span></code></pre></div>
<ul>
<li>Again, this 95% CI does not encompass 0; therefore, the null hypothesis can be rejected at this level</li>
<li>In the previous <code><a href="https://rdrr.io/r/base/summary.html">summary()</a></code> chunk, the standard two sample t-test was performed without factoring in the pairing</li>
</ul>
<p>Generate table of coefficients for a paired t-test:</p>
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">height</span> <span class="op">~</span> <span class="va">type</span> <span class="op">+</span> <span class="va">pair</span>, data <span class="op">=</span> <span class="va">darwin</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:</span>
<span class="co">#&gt; lm(formula = height ~ type + pair, data = darwin)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Residuals:</span>
<span class="co">#&gt;     Min      1Q  Median      3Q     Max </span>
<span class="co">#&gt; -5.4958 -0.9021  0.0000  0.9021  5.4958 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Coefficients:</span>
<span class="co">#&gt;             Estimate Std. Error t value Pr(&gt;|t|)    </span>
<span class="co">#&gt; (Intercept)  21.7458     2.4364   8.925 3.75e-07 ***</span>
<span class="co">#&gt; typeSelf     -2.6167     1.2182  -2.148   0.0497 *  </span>
<span class="co">#&gt; pair2        -4.2500     3.3362  -1.274   0.2234    </span>
<span class="co">#&gt; pair3         0.0625     3.3362   0.019   0.9853    </span>
<span class="co">#&gt; pair4         0.5625     3.3362   0.169   0.8685    </span>
<span class="co">#&gt; pair5        -1.6875     3.3362  -0.506   0.6209    </span>
<span class="co">#&gt; pair6        -0.3750     3.3362  -0.112   0.9121    </span>
<span class="co">#&gt; pair7        -0.0625     3.3362  -0.019   0.9853    </span>
<span class="co">#&gt; pair8        -2.6250     3.3362  -0.787   0.4445    </span>
<span class="co">#&gt; pair9        -3.0625     3.3362  -0.918   0.3742    </span>
<span class="co">#&gt; pair10       -0.6250     3.3362  -0.187   0.8541    </span>
<span class="co">#&gt; pair11       -0.6875     3.3362  -0.206   0.8397    </span>
<span class="co">#&gt; pair12       -0.9375     3.3362  -0.281   0.7828    </span>
<span class="co">#&gt; pair13       -3.0000     3.3362  -0.899   0.3837    </span>
<span class="co">#&gt; pair14       -1.1875     3.3362  -0.356   0.7272    </span>
<span class="co">#&gt; pair15       -5.4375     3.3362  -1.630   0.1254    </span>
<span class="co">#&gt; ---</span>
<span class="co">#&gt; Signif. codes:  </span>
<span class="co">#&gt; 0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Residual standard error: 3.336 on 14 degrees of freedom</span>
<span class="co">#&gt; Multiple R-squared:  0.469,  Adjusted R-squared:  -0.09997 </span>
<span class="co">#&gt; F-statistic: 0.8243 on 15 and 14 DF,  p-value: 0.6434</span></code></pre></div>
<ul>
<li>New table of coefficients output:
<ul>
<li>Takes the crossed plant from pair 1 as the intercept</li>
<li>Shows the mean difference in height of the crossed plants (-2.6167)</li>
<li>Then shows the average difference of each pair relative to pair 1</li>
</ul>
</li>
<li>Ignore these differences of pairs and see that the second row gives us the t value for the paired t-test:</li>
</ul>
<div class="sourceCode" id="cb112"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">-</span><span class="fl">2.6167</span><span class="op">/</span><span class="fl">1.2182</span>
<span class="co">#&gt; [1] -2.148005</span></code></pre></div>
<ul>
<li>This is the observed t value, which is way smaller than the critical t value (2.144787)</li>
<li>This finding corresponds to the large p value in the table which is 0.6434</li>
</ul>
<p>We can use a the linear model function to perform the equivalent of a one-sample t test:
- generate a single sample of differences:</p>
<div class="sourceCode" id="cb113"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ex0428</span><span class="op">$</span><span class="va">Difference</span> <span class="op">&lt;-</span> <span class="va">ex0428</span><span class="op">$</span><span class="va">Cross</span> <span class="op">-</span> <span class="va">ex0428</span><span class="op">$</span><span class="va">Self</span></code></pre></div>
<p>Fit a linear model that only estimates the mean difference:</p>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">Difference</span> <span class="op">~</span><span class="fl">1</span>, <span class="co">#the 1 indicates the intercept, which is the mean of a single sample of differences</span>
           data <span class="op">=</span> <span class="va">ex0428</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:</span>
<span class="co">#&gt; lm(formula = Difference ~ 1, data = ex0428)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Residuals:</span>
<span class="co">#&gt;      Min       1Q   Median       3Q      Max </span>
<span class="co">#&gt; -10.9917  -1.2417   0.3833   3.0083   6.7583 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Coefficients:</span>
<span class="co">#&gt;             Estimate Std. Error t value Pr(&gt;|t|)  </span>
<span class="co">#&gt; (Intercept)    2.617      1.218   2.148   0.0497 *</span>
<span class="co">#&gt; ---</span>
<span class="co">#&gt; Signif. codes:  </span>
<span class="co">#&gt; 0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Residual standard error: 4.718 on 14 degrees of freedom</span></code></pre></div>
</div>
<div id="summary-statistics-1" class="section level2" number="9.3">
<h2>
<span class="header-section-number">9.3</span> Summary: statistics<a class="anchor" aria-label="anchor" href="#summary-statistics-1"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li><a href="https://www.scribbr.com/statistics/t-test/">Alternate source on t-tests</a></li>
<li>I will probably rarely if ever use a one sample t-test</li>
<li>use two-tailed t-test since we don’t need to determine the direction of the difference</li>
<li>use a paired t-test if the groups come from a single population
<ul>
<li>i.e. if you are measuring gene expression in two groups of cannabis plants over time or before/after methyl jasmonate treatment</li>
</ul>
</li>
<li>use a two sample (unpaired) t-test if the two groups come from different populations
<ul>
<li>i.e. comparing some feature of two varieties of P. cubensis</li>
<li>if they are different, then the CI will not include 0 so the null hypothesis can be rejected</li>
</ul>
</li>
</ul>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="prediction.html"><span class="header-section-number">8</span> Prediction</a></div>
<div class="next"><a href="intervals.html"><span class="header-section-number">10</span> Intervals</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#testing"><span class="header-section-number">9</span> Testing</a></li>
<li><a class="nav-link" href="#significance-testing-time-for-t"><span class="header-section-number">9.1</span> Significance testing: Time for t</a></li>
<li><a class="nav-link" href="#students-t-test-darwins-maize"><span class="header-section-number">9.2</span> Student’s t-test: Darwin’s maize</a></li>
<li><a class="nav-link" href="#summary-statistics-1"><span class="header-section-number">9.3</span> Summary: statistics</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/apicellap/NSwR.git/blob/master/09_testing.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/apicellap/NSwR.git/edit/master/09_testing.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>A Guide to The New Statistics with R</strong>" was written by Pete Apicella. It was last built on 2022-07-19.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
